frontend/src/components/ViewProfil/ViewProfil.jsx

```diff
@@ -26,70 +28,122 @@ export default function ViewProfil() {
     });
   };
 
+  const hAvatarChoice = (evt) => {
+    setUserProfil({
+      ...userProfil,
+      path: evt.target.src,
+    });
+    Query.editUser(
+      {
+        ...userProfil,
+        [evt.target.name]: parseInt(evt.target.id, 10),
+      },
+      currentUser.id
+    );
+    setTimeout(() => toggleAvatarsShow(), 1000);
+    swal({
+      title: "Done",
+      text: "Avatar changed",
+      icon: "success",
+      buttons: false,
+      timer: 1000,
+    });
+  };
+
   return (
-    <form className="blocProfil" onSubmit={hEditFormSubmit}>
-      <header>
-        <img className="avatar" src={avatar1} alt="avatar profil" />
-        <h1 key={currentUser.id}>Hello {currentUser.username} </h1>
-      </header>
-      <div className="inputProf">
-        <p className="viewProfil">
-          Pseudo
-          <input
-            type="text"
-            name="username"
-            value={userProfil.username}
-            onChange={hFormChange}
-          />
-        </p>
-        <p className="viewProfil">
-          Prénom
-          <input
-            type="text"
-            name="firstname"
-            value={userProfil.firstname}
-            onChange={hFormChange}
-          />
-        </p>
-        <p className="viewProfil">
-          Nom
-          <input
-            type="text"
-            name="lastname"
-            value={userProfil.lastname}
-            onChange={hFormChange}
+    userProfil && (
+      <form className="blocProfil" onSubmit={hEditFormSubmit}>
+        <Modal
+          isShowing={showAvatars}
+          hide={toggleAvatarsShow}
+          title="Choose your avatar"
+        >
+          <div className="avatars">
+            {avatars.map((avatar) => {
+              return (
+                <img
+                  className="avatar"
+                  src={avatar.path.replace(`/http://localhost:3000/g`, "")}
+                  alt="avatar choice"
+                  key={avatar.id}
+                  id={avatar.id}
+                  onClick={hAvatarChoice}
+                  role="presentation"
+                  name="avatar_id"
+                />
+              );
+            })}
+          </div>
+        </Modal>
+        <header>
+          <h1 key={currentUser.id}>Hello {currentUser.username} </h1>
+          <img
+            className="avatar"
+            src={userProfil.path.replace(`/http://localhost:3000/g`, "")}
```

Comment:
c'est quoi ce /g ?

Reply:
(Without the backticks I'd have guessed it's the "greedy" flag, but in this case I have no idea either)

Reply:
pour la regex c'est le flag global

Reply:
et oui j'ai oublié de changer le nom de class (j'arrive pas a répondre sous l'autre post)